AGENTS.md

```diff
@@ -0,0 +1,577 @@
+)
+
+try:
+    result = agent("Complex query...")
+except ContextWindowOverflowException:
+    # Handle context overflow
+    agent.conversation_manager.clear_history()
+except ModelThrottledException:
+    # Handle rate limiting
+    time.sleep(60)
+```
+
+### Performance Optimization
+
+1. **Use streaming for long responses**:
+   ```python
+   async for chunk in agent.stream("Generate a long report..."):
+       print(chunk.content, end="", flush=True)
+   ```
+
+2. **Configure appropriate context windows**:
+   ```python
+   from strands.agent.conversation_manager import SlidingWindowConversationManager
+   
+   conversation_manager = SlidingWindowConversationManager(max_messages=20)
+   agent = Agent(conversation_manager=conversation_manager)
+   ```
+
+3. **Use concurrent execution for independent tools**:
+   ```python
+   from strands.tools.executors import ConcurrentToolExecutor
+   
+   agent = Agent(tool_executor=ConcurrentToolExecutor(max_workers=4))
+   ```
+
+---
+
+## Testing and Development
+
+### Integration Test Patterns
+
+The codebase includes comprehensive integration tests demonstrating real usage:
+
+- **`test_function_tools.py`**: Basic tool creation and usage
+- **`test_multiagent_swarm.py`**: Multi-agent coordination
+- **`test_mcp_client.py`**: MCP server integration
+- **`test_stream_agent.py`**: Streaming responses
+- **`test_session.py`**: Session persistence
+
+### Development Environment Setup
+
+```bash
+# Create virtual environment
+python -m venv .venv
+source .venv/bin/activate
+
+# Install with development dependencies
+pip install -e ".[dev,all]"
+
+# Run tests
+hatch test                    # Unit tests
+hatch test tests_integ       # Integration tests
+
+# Code quality
+hatch fmt --formatter        # Format code
+hatch fmt --linter          # Lint code
+```
+
+---
+
+## Configuration Reference
+
+### Model Provider Configurations
+
+**AWS Bedrock:**
+```python
+BedrockModel(
+    model_id="us.amazon.nova-pro-v1:0",
+    temperature=0.3,
+    max_tokens=4096,
+    top_p=0.9,
+    streaming=True,
+    guardrails_config={...}
+)
+```
+
+**OpenAI:**
+```python
+OpenAIModel(
+    model_id="gpt-4",
+    api_key="sk-...",
+    temperature=0.7,
+    max_tokens=2048,
+    organization="org-..."
+)
+```
+
+**Anthropic:**
+```python
+AnthropicModel(
+    model_id="claude-3-sonnet-20240229",
+    api_key="sk-ant-...",
+    temperature=0.5,
+    max_tokens=4096
+)
+```
+
+### Tool Executor Options
+
+```python
+# Sequential execution (default)
+from strands.tools.executors import SequentialToolExecutor
+agent = Agent(tool_executor=SequentialToolExecutor())
+
+# Concurrent execution
+from strands.tools.executors import ConcurrentToolExecutor
+agent = Agent(tool_executor=ConcurrentToolExecutor(max_workers=4))
+```
+
+---
+
+## Troubleshooting Guide
+
+### Common Issues and Solutions
+
+1. **Context Window Overflow**:
+   - Use `SlidingWindowConversationManager`
+   - Implement conversation summarization
```

Comment:
We don't recommend these last two points at all.

And "implement conversation summarization" - isn't that https://strandsagents.com/latest/documentation/docs/user-guide/concepts/agents/conversation-management/#summarizingconversationmanager?